cluster.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const process = require('process');
+const cluster = require('cluster');
+const { config } = require('./constants.js');
+const { watch } = require('fs');
+const logger = require('./loggerFactory.js');
+const Proxy = require('./Proxy.js');
+const ProcessHandler = require('./ProcessHandler.js');
+
+const WORKER_CNT = config.workerVal;
+const activeWorkers = [];
+
+function disconnectOldWorkers() {
+	const len = activeWorkers.length;
+	for (let i = 0; i < len; ++i) {
+		const oldWorker = activeWorkers.shift();
+		oldWorker.disconnect();
+		setTimeout(() => oldWorker.kill(), 60 * 10 * 1000);
+	}
+}
+
+function spawnNewWorkers() {
+	if (cluster.isMaster) {
+		for (let i = 0; i < WORKER_CNT; ++i) {
+			const worker = cluster.fork();
+			worker.on('error', (err) => {
+				logger.error(`Received error event on ${worker.id} : ${err}`);
+			});
+			logger.info(`Created worker with id ${worker.id}`);
+			activeWorkers.push(worker);
+		}
+	}
+}
+
+let currTime = Date.now();
+watch('tmp/restart.txt', () => {
```

Comment:
please create `logs` directory if not exists. when code starts, throws exception when run `npm start`. also create a`tmp/restart.txt`

Reply:
added

Reply:
we are removing from `.gitignore`. this can make it difficult to develop. logs and any files in tmp, will appear in git staging. instead of removing from `.gitignore`, can we ensure that they exist before use?

Reply:
Have added back in .gitignore

Reply:
this should be inside master or will be watched by each worker process.

Reply:
updated